.github/workflows/image.yaml

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+---
+name: Build
+
+"on":
+  push:
+    branches:
+      - main
+
+jobs:
+  Image-build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: build image
+        id: build-image
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: enterprise-contract
+          tags: latest ${{ github.sha }}
```

Comment:
There is a race condition between when this is run and the state of the git repository, the commit id tag might be placed on the wrong image version as a result. See below for incorporating this into the release workflow.